go/tools/bazel/runfiles/runfiles.go

```diff
@@ -0,0 +1,201 @@
+// of a runfile, and use Env to obtain environmental variables to pass on to
+// subprocesses.
+//
+// Runfiles type
+//
+// If you need hermetic behavior or want to change the runfiles discovery
+// process, use New to create a Runfiles object.  New accepts a few options to
+// change the discovery process.  Runfiles objects have methods Path and Env,
+// which correspond to the package-level functions.  On Go 1.16, *Runfiles
+// implements fs.FS, fs.StatFS, and fs.ReadFileFS.
+package runfiles
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	directoryVar    = "RUNFILES_DIR"
+	manifestFileVar = "RUNFILES_MANIFEST_FILE"
+)
+
+// Runfiles allows access to Bazel runfiles.  Use New to create Runfiles
+// objects; the zero Runfiles object always returns errors.  See
+// https://docs.bazel.build/skylark/rules.html#runfiles for some information on
+// Bazel runfiles.
+type Runfiles struct {
+	// We don’t need concurrency control since Runfiles objects are
+	// immutable once created.
+	impl runfiles
+	env  string
+}
+
+// New creates a given Runfiles object.  By default, it uses os.Args and the
+// RUNFILES_MANIFEST_FILE and RUNFILES_DIR environmental variables to find the
+// runfiles location.  This can be overwritten by passing some options.
+//
+// See section “Runfiles discovery” in
+// https://docs.google.com/document/d/e/2PACX-1vSDIrFnFvEYhKsCMdGdD40wZRBX3m3aZ5HhVj4CtHPmiXKDCxioTUbYsDydjKtFDAzER5eg7OjJWs3V/pub.
+func New(opts ...Option) (*Runfiles, error) {
+	var o options
+	for _, a := range opts {
+		a.apply(&o)
+	}
+
+	if o.manifest == "" {
+		o.manifest = ManifestFile(os.Getenv(manifestFileVar))
+	}
+	if o.manifest != "" {
+		return o.manifest.new()
+	}
+
+	if o.directory == "" {
+		o.directory = Directory(os.Getenv(directoryVar))
+	}
+	if o.directory != "" {
+		return o.directory.new(), nil
+	}
+
+	if o.program == "" {
+		o.program = ProgramName(os.Args[0])
+	}
+	manifest := ManifestFile(o.program + ".runfiles_manifest")
+	if stat, err := os.Stat(string(manifest)); err == nil && stat.Mode().IsRegular() {
+		return manifest.new()
+	}
+
+	dir := Directory(o.program + ".runfiles")
+	if stat, err := os.Stat(string(dir)); err == nil && stat.IsDir() {
+		return dir.new(), nil
+	}
+
+	return nil, errors.New("runfiles: no runfiles found")
+}
+
+// Path returns the absolute path name of a runfile.  The runfile name must be a
+// runfile-root relative path, using the slash (not backslash) as directory separator.
+// If r is the zero Runfiles object, Path always returns an error.  If the runfiles
+// manifest maps s to an empty name (indicating an empty runfile not present in the
+// filesystem), Path returns an error that wraps ErrEmpty.
+//
+// See section “Library interface” in
+// https://docs.google.com/document/d/e/2PACX-1vSDIrFnFvEYhKsCMdGdD40wZRBX3m3aZ5HhVj4CtHPmiXKDCxioTUbYsDydjKtFDAzER5eg7OjJWs3V/pub.
+func (r *Runfiles) Path(s string) (string, error) {
```

Comment:
```suggestion
func (r *Runfiles) Path(relPath string) (string, error) {
```

Reply:
The function allows absolute paths as well, so I am going to use `path` instead and adding an example.